catalog/solr/catalog-solr-external-provider/src/main/java/ddf/catalog/solr/external/SolrHttpCatalogProvider.java

```diff
@@ -231,43 +236,46 @@ public void updateServer(String urlValue) {
         LOGGER.info("New url {}", urlValue);
 
         if (urlValue != null) {
-            if (!StringUtils.equalsIgnoreCase(urlValue.trim(), url) || server == null) {
+            if (!StringUtils.equalsIgnoreCase(urlValue.trim(), url) || getServer() == null) {
                 url = urlValue.trim();
 
-                if (server != null) {
+                if (getServer() != null) {
                     LOGGER.info(
                             "Shutting down the connection manager to the Solr Server and releasing allocated resources.");
-                    server.shutdown();
+                    getServer().shutdown();
                     LOGGER.info("Shutdown complete.");
                 }
 
-                server = SolrServerFactory
-                        .getHttpSolrServer(url, SOLR_CATALOG_CORE_NAME, SOLR_CATALOG_CONFIG_FILE);
-
-                firstUse = true;
+                updateServer();
             }
+
         } else {
             // sets to null
             url = null;
         }
     }
 
+    private void updateServer() {
+        serverFuture = SolrServerFactory.getHttpSolrServer(url,
+                SOLR_CATALOG_CORE_NAME,
+                SOLR_CATALOG_CONFIG_FILE);
+        server = null;
+    }
+
     private CatalogProvider getProvider() {
-        if (firstUse) {
-            if (isServerUp(this.server)) {
-                if (resolver == null) {
-                    provider = new SolrCatalogProvider(server, filterAdapter,
-                            solrFilterDelegateFactory);
-                } else {
-                    provider = new SolrCatalogProvider(server, filterAdapter,
-                            solrFilterDelegateFactory, resolver);
-                }
-                provider.maskId(getId());
-                this.firstUse = false;
-                return provider;
-            }
+
+        if (!isServerUp(getServer())) {
             return new UnconfiguredCatalogProvider();
         }
+
+        if (provider == null || provider instanceof UnconfiguredCatalogProvider) {
```

Comment:
Yeah, it looks like the null checks in this file might be redundant now. They *were* applicable before this PR. :smile: